lib/strategies/webapp-strategy.js

```diff
@@ -207,6 +208,36 @@ function handleAuthorization(req, options, strategy) {
 		options.successRedirect = req.url;
 	}
 
+	if (typeof(strategy.getRefreshToken) === "function") {
+		return obtainTokensWithRefreshToken(req, options, strategy).catch(function() {
+			redirectForAuthorization(req, options, strategy);
```

Comment:
Would be nice to have a debug here that tells us why did we continue to authorization (no refresh token returned from user or refresh token failed on server?)